src/eventshub/src/component/Search/Middleware.tsx

```diff
@@ -20,7 +20,7 @@ export type RouteState = Partial<{
  * Callback function --------------------------------------------
  */
 function middleware({ instantSearchInstance }: { instantSearchInstance: InstantSearch<ExtendedUIState> }) {
-  const hi = history<RouteState>({cleanUrlOnDispose: false,});
+  const hi = history<RouteState>({ cleanUrlOnDispose: false, writeDelay: 10 });
```

Comment:
Short delay to make it appear more instant but enough to catch rapid state changes